webapi/Controllers/DocumentController.cs

```diff
@@ -136,20 +142,22 @@ public class DocumentController : ControllerBase
         // Pre-create chat-message
         DocumentMessageContent documentMessageContent = new();
 
-        var importResults = await ImportDocumentsAsync();
+        var importResults = await this.ImportDocumentsAsync(memoryClient, chatId, documentImportForm, documentMessageContent);
+
+        var chatMessage = await this.TryCreateDocumentUploadMessage(chatId, documentMessageContent);
 
         // Broadcast the document uploaded event to other users.
         if (documentScope == DocumentScopes.Chat)
         {
-            var chatMessage = await this.TryCreateDocumentUploadMessage(chatId, documentMessageContent);
-
             // If chat message isn't created, it is still broadcast and visible in the documents tab.
             // The chat message won't, however, be displayed when the chat is freshly rendered.
 
             var userId = this._authInfo.UserId;
             await messageRelayHubContext.Clients.Group(chatId.ToString())
                 .SendAsync(ReceiveMessageClientCall, chatId, userId, chatMessage);
 
+            this._logger.LogInformation("Local upload chat message: {0}", chatMessage!.ToString());
```

Comment:
Only ever user the '!' operator when you are 100% certain it can't be null - In this case it actually can!

We probably want to return above (likely a 400) if chatMessage is null.

Then the '!' won't be needed.  Same with the other log.